### ISSUE TITLE
Wagtail Datepicker not showing on custom filterset_class for DateField (no JS error)

### ISSUE THREAD

**Issue by Quadric** [wagtail/wagtail#12225]
### Issue Summary

I though it's the same problem as in https://github.com/wagtail/wagtail/issues/10254 but as I understand it is fixed in Wagtail 5 but I'm using Wagtail 5.2. I have no JS error in console but DatePicker is not showing up

### Steps to Reproduce

I have created a custom filterset like this:
```python
import django_filters
from django.utils.translation import gettext_lazy as _
from wagtail.admin.filters import WagtailFilterSet

from .models import FormSubmission


class FormSubmissionFilter(WagtailFilterSet):
    created_at_date = django_filters.DateFilter(
        field_name="created_at", lookup_expr="date", label=_("creation date")
    )

    class Meta:
        model = FormSubmission
        fields = ["created_at_date"]
```
and according to implementation in WagtailFilterSet it should detect that my field is DateField and it should change automatically widget to `AdminDateInput`.

```python
class WagtailFilterSet(django_filters.FilterSet):
    @classmethod
    def filter_for_lookup(cls, field, lookup_type):
        filter_class, params = super().filter_for_lookup(field, lookup_type)

        if filter_class == django_filters.ChoiceFilter:
            params.setdefault("widget", forms.RadioSelect)
            params.setdefault("empty_label", _("All"))

        elif filter_class in [django_filters.DateFilter, django_filters.DateTimeFilter]:
            params.setdefault("widget", AdminDateInput)

        elif filter_class == django_filters.DateFromToRangeFilter:
            params.setdefault("widget", DateRangePickerWidget)

        elif filter_class == django_filters.BooleanFilter:
            params.setdefault("widget", BooleanRadioSelect)

        return filter_class, params
```
But It does not.

![image](https://github.com/user-attachments/assets/9d6e4cf4-4421-4e56-ac5c-d1e7a770faf8)

If I add manually this widget `AdminDateInput` then it works and DatePicker shows

```python
class FormSubmissionFilter(WagtailFilterSet):
    created_at_date = django_filters.DateFilter(
        field_name="created_at", lookup_expr="date", label=_("creation date"), widget=AdminDateInput()
    )

    class Meta:
        model = FormSubmission
        fields = ["created_at_date"]
```

![image](https://github.com/user-attachments/assets/4da332dc-6c20-4bb5-bb76-a184bbbbb40b)

### Any other relevant information. For example, why do you consider this a bug and what did you expect to happen instead?
I guess having `WagtailFilterSet` as a parent for ma FilterSet should make my DateField having a DatePicker compatible with Wagtail

### Technical details

Python version: 3.11.9
Django version: 4.2.13
Wagtail version: 5.2.6
Browser version: Edge 127

**Comment by laymonage** [comment]
django-filter `FilterSet`s work like `ModelForm`s, i.e. it auto-generates the filters based on the `Meta.model` and `Meta.fields`, but if you declare a filter with the same name on the class itself, that filter takes precedence. That means adding `fields = ["created_at_date"]` in your filterset's `Meta` class has no effect. The `filter_for_lookup` method is only used when auto-generating the filters based on `fields`.

In short, either remove your explicit `created_at_date` filter definition and let django-filter auto-generate it from `Meta.fields`, or specify the `widget` in your filter definition.

**Comment by Quadric** [comment]
Hey @laymonage thank you for your reply.

1. removing "created_at_date" from fields of Filterset does not help
2. I could autogenerate this filter from modeladmin but it makes "auto generated" name which is not acceptable. I read documentation of django-filter and examples with customizing name of filter were only when filter definition was manullay created.

Eventually I need a filter that will gives me the ability to filter my models' "created_at" datetime by "date".

I guess you mean doing something like this:

```python
class SubmissionListView(IndexView):
    model = FormSubmission
    queryset = FormSubmission.objects.all()
    pk_url_kwarg = "pk"
    # filterset_class = FormSubmissionFilter
    list_filter = {
        "created_at": ("date",)
    }
```
But it gives me 
![image](https://github.com/user-attachments/assets/b58c8713-9e2d-4d2e-86ac-3961c39cf5cc)
And it looks and sound OK - in english... but in polish

![image](https://github.com/user-attachments/assets/edbfdd5b-f7dc-455f-8c62-b10e25ddc84c)

this label does not make sense - it's mixed polish and english.

So using Wagtail and Filterset abilities of using Model defiinition of created_at how can I just change the label for this filter?

**Comment by laymonage** [comment]
If you want to customise the label, you need a custom filterset class. Either use what you have and specify `widget`, or like `ModelForm`s, you can modify stuff in `__init__`, e.g.

```py
class FormSubmissionFilter(WagtailFilterSet):
    def __init__(self, data=None, queryset=None, *, request=None, prefix=None):
        super().__init__(data, queryset, request=request, prefix=prefix)
        self.filters["created_at__date"].label = gettext("creation date")

    class Meta:
        model = FormSubmission
        fields = {
            "created_at": ("date",)
        }
```
